lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,565 @@
+        loss = torch.stack([x["loss"] for x in outputs]).mean()
+        self.log("train_loss", loss, prog_bar=True, sync_dist=True)
+        if self.finetuningscheduler_callback:
+            self.log("finetuning_schedule_depth", float(self.finetuningscheduler_callback.curr_depth))
+
+    def validation_step(self, batch, batch_idx, dataloader_idx=0):
+        outputs = self(**batch)
+        val_loss, logits = outputs[:2]
+
+        if self.num_labels >= 1:
+            preds = torch.argmax(logits, axis=1)
+        elif self.num_labels == 1:
+            preds = logits.squeeze()
+
+        labels = batch["labels"]
+        return {"loss": val_loss, "preds": preds, "labels": labels}
+
+    def validation_epoch_end(self, outputs):
+        preds = torch.cat([x["preds"] for x in outputs]).detach().cpu().numpy()
+        labels = torch.cat([x["labels"] for x in outputs]).detach().cpu().numpy()
+        loss = torch.stack([x["loss"] for x in outputs]).mean()
+        self.log("val_loss", loss, prog_bar=True, sync_dist=True)
+        metric_dict = self.metric.compute(predictions=preds, references=labels)
+        self.log_dict(metric_dict, prog_bar=True, sync_dist=True)
+        return loss
+
+    def init_pgs(self) -> List[Dict]:
+        """Initialize the parameter groups. Used to ensure weight_decay is not applied to our specified bias
+        parameters when we initialize the optimizer.
+
+        Returns:
+            List[Dict]: A list of parameter group dictionaries.
+        """
+        pgs = [
+            {
+                "params": [
+                    p
+                    for n, p in self.model.named_parameters()
+                    if not any(nd in n for nd in self.no_decay) and p.requires_grad
+                ],
+                "weight_decay": self.optimizer_init["init_args"]["weight_decay"],
+            },
+            {
+                "params": [
+                    p
+                    for n, p in self.model.named_parameters()
+                    if any(nd in n for nd in self.no_decay) and p.requires_grad
+                ],
+                "weight_decay": 0.0,
+            },
+        ]
+        return pgs
+
+    def configure_optimizers(self):
+        # the phase 0 parameters will have been set to require gradients during setup
+        # you can initialize the optimizer with a simple requires.grad filter as is often done,
+        # but in this case we pass a list of parameter groups to ensure weight_decay is
+        # not applied to the bias parameter (for completeness, in this case it won't make much
+        # performance difference)
+        optimizer = instantiate_registered_class(args=self.init_pgs(), init=self.optimizer_init)
+        scheduler = {
+            "scheduler": instantiate_registered_class(args=optimizer, init=self.lr_scheduler_init),
+            **self.pl_lrs_cfg,
+        }
+        return [optimizer], [scheduler]
+
+    def configure_callbacks(self):
```

Comment:
Neat !

Reply:
Might be better to move this as a property. This isn't really the desired behaviour. `configure_callbacks` is meant to define a callback associated to a LightningModule, so they get injected directly within the Trainer.

Reply:
Agreed! (if I'm understanding you correctly :) ). I originally had included a finetuning_scheduler_callback property (analogous to the early_stopping_callback property) in the  [FinetuningScheduler PR](https://github.com/PyTorchLightning/pytorch-lightning/pull/10198) that contains a few minor changes to ``Trainer`` to facilitate the finetuning scheduler functionality. 

I decided to remove the property to keep the core pl changes to an absolute minimum. Right now, the only trainer.py change is the addition of the self.fit_ckpt_path attribute so that the ckpt_path can be accessed/mutated within the context within the context of fit.

Should I re-add the finetuning_scheduler_callback property back to the aforementioned PR?